ext/nokogiri/xml_sax_parser.c

```diff
@@ -203,10 +203,16 @@ warning_func(void *ctx, const char *msg, ...)
   VALUE doc = rb_iv_get(self, "@document");
   VALUE rb_message;
 
+#ifdef TRUFFLERUBY_NOKOGIRI_SYSTEM_LIBRARIES
+  /* It is not currently possible to pass var args from native
+     functions to sulong, so we work around the issue here. */
```

Comment:
I've asked the Sulong team about that (@rschatz), and it sounds like a limitation of libffi (can't read varargs from a libffi callback), so probably nothing we can do about it in the short term.

Reply:
Links I've found about that from a quick search:
* https://github.com/libffi/libffi/issues/388
* https://github.com/frida/frida-gum/issues/364#issuecomment-547761075